apiary-metastore-listener/src/main/java/com/expedia/apiary/extensions/metastore/listener/ApiarySnsListener.java

```diff
@@ -149,10 +150,13 @@ private void publishEvent(
     }
     if (partition != null) {
       JSONArray partitionValuesArray = new JSONArray(partition.getValues());
-      JSONArray partitionKeysArray = new JSONArray(
-          table.getPartitionKeys().stream().map(f -> f.getName()).collect(Collectors.toList()));
+      Map<String, String> partitionKeysMap = new LinkedHashMap<>();
```

Comment:
This is one of the _very few_ cases where I think it's actually better to have the concrete type (`LinkedHashMap`) on the LHS as it really is important to make clear that order matters in this case.